sdl2-sys/build.rs

```diff
@@ -1,19 +1,119 @@
 #[cfg(feature="pkg-config")]
 extern crate pkg_config;
+#[cfg(feature="bundled")]
+extern crate cmake;
+#[cfg(feature="bundled")]
+extern crate tar;
+#[cfg(feature="bundled")]
+extern crate flate2;
+#[cfg(feature="bundled")]
+extern crate reqwest;
 
+use std::{io, env};
+use std::path::Path;
+use std::fs;
+
+struct TargetInfo {
+    _triple: String,
+    os: String,
+}
+
+impl TargetInfo {
+    fn init() -> TargetInfo {
+        let triple = ::std::env::var("TARGET").expect("Cargo build scripts always have TARGET");
+        let os = triple.splitn(3, "-").nth(2).unwrap().to_string();
+        TargetInfo {
+            _triple: triple,
+            os: os,
+        }
+    }
+}
+
+#[cfg(feature="bundled")]
+fn download_to<T: io::Write>(url: &str, mut dest: T) {
+    use io::BufRead;
+
+    let resp = reqwest::get(url).expect(&format!("Failed to GET resource: {:?}", url));
+    let size = resp.headers()
+        .get::<reqwest::header::ContentLength>()
+        .map(|ct_len| **ct_len)
+        .unwrap_or(0);
+    if !resp.status().is_success() { panic!("Download request failed with status: {:?}", resp.status()) }
+    if size == 0 { panic!("Size of content was returned was 0") }
+
+    let mut src = io::BufReader::new(resp);
+    loop {
+        let n = {
+            let mut buf = src.fill_buf().unwrap();
+            dest.write_all(&mut buf).unwrap();
+            buf.len()
+        };
+        if n == 0 { break; }
+        src.consume(n);
+    }
+}
+
+#[cfg(feature="bundled")]
+fn main() {
+    let target_info = TargetInfo::init();
+    let sdl2_archive_url = "http://libsdl.org/release/SDL2-2.0.5.tar.gz";
```

Comment:
Can you put "2.0.5" in a `const` variable somwhere and use the const variable everytime you have "2.0.5"? It will be nicer to edit a single const from "2.0.5" to "2.0.6" instead of searching for all the lines.